### PR TITLE
Iterator filter predicates for boundary and manifold IDs

### PR DESCRIPTION
**Pull request by jppelteret**

### CODE REVIEW

File: tests/grid/filtered_iterator_06_operator.cc [hunk 1]
```diff
@@ -51,28 +52,47 @@ test()
...
+            boundary_face_set.insert(cell->face(face_n));
+
+          if (cell->face(face_n)->manifold_id() == manifold_id)
+            manifold_face_set.insert(cell->face(face_n));
+        }
+
+  std::size_t n_boundary_filtered_cells = 0;
+  for (const auto &filtered_cell :
+       tria.active_face_iterators() | IteratorFilters::AtBoundary() |
+         IteratorFilters::BoundaryIdEqualTo(boundary_id))
```
Comment (bangerth) [inline tests/grid/filtered_iterator_06_operator.cc:76]: :-) That's just elegant!

**Review by bangerth — APPROVED** [review]

File: tests/grid/filtered_iterator_06_operator.cc (hunk 1 above)
Comment (jppelteret) [reply tests/grid/filtered_iterator_06_operator.cc:76]: Yeah, I'm really happy that you implemented `operator|`. It's so nice to chain these filters!

Comment (bangerth) [reply tests/grid/filtered_iterator_06_operator.cc:76]: I really like that syntax. It's been used on the unix command line for 50 years, of course, and I like that we'll be able to use it in C++ as well (or at least will be with C++20).

**Pull request merged**